tests/infra/piccolo_driver.py

```diff
@@ -0,0 +1,315 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import argparse
+import os
+import infra.e2e_args
+import infra.remote_client
+from infra.perf import PERF_COLUMNS
+from random import seed
+import getpass
+from loguru import logger as LOG
+import cimetrics.upload
+import time
+import http
+import sys
+
+sys.path.insert(0, "../tests/perf-system/generator")
+import generator
+
+sys.path.insert(0, "../tests/perf-system/analyzer")
+import analyzer
+
+
+def get_command_args(args, get_command):
+    command_args = []
+    return get_command(*command_args)
+
+
+def minimum_number_of_local_nodes(args):
+    """
+    If we are using bft then we need to have 3 nodes. CFT will run with 1 nodes, unless it expects a backup
+    """
+    if args.consensus == "BFT":
+        return 3
+
+    if args.send_tx_to == "backups":
+        return 2
+
+    return 1
+
+
+def filter_nodes(primary, backups, filter_type):
+    if filter_type == "primary":
+        return [primary]
+    elif filter_type == "backups":
+        if not backups:
+            raise Exception("--send-tx-to backups but no backup was found")
+        return backups
+    else:
+        return [primary] + backups
+
+
+def my_configure_remote_client(args, client_id, client_host, node, command_args):
+    if client_host == "localhost":
+        client_host = infra.net.expand_localhost()
+        remote_impl = infra.remote.LocalRemote
+    else:
+        remote_impl = infra.remote.SSHRemote
+    try:
+        remote_client = infra.remote_client.CCFRemoteClient(
+            "client_" + str(client_id),
+            client_host,
+            args.client,
+            node.get_public_rpc_host(),
+            node.get_public_rpc_port(),
+            args.workspace,
+            args.label,
+            args.config,
+            command_args,
+            remote_impl,
+            piccolo_run=True,
+        )
+        remote_client.setup()
+        return remote_client
+    except Exception:
+        LOG.exception("Failed to start client {}".format(client_host))
+        raise
+
+
+def run(get_command, args):
+    if args.fixed_seed:
+        seed(getpass.getuser())
+
+    hosts = args.nodes
+    if not hosts:
+        hosts = ["local://localhost"] * minimum_number_of_local_nodes(args)
+
+    args.initial_user_count = 3
+    args.sig_ms_interval = 1000  # Set to cchost default value
```

Comment:
```suggestion
    args.sig_ms_interval = 100
```

We don't want 1 second delay at the end of the test if we don't end up on a round 1000 txs, which we probably won't.